### PR TITLE
Time series updates

### PR DESCRIPTION
**Pull request by daniel-thom**
This updates PowerSystems to with the pre-10 branch of InfrastructureSystems.  It also requires the pre-10 branch of PowerSystemsTestData.

A number of things are commented out.  Leaving this as a draft until we workout what to do with them.

All tests pass except for 
- Some in test_read_time_series.jl.  These failures are due to an interface change in IS.read_time_series.  We currently don't have a way to convert PSY-supported time series CSV files into TimeArray objects.  Leaving this unresolved until we determine whether we want to restore the prior functionality.

### CODE REVIEW

File: src/base.jl [hunk 1]
```diff
@@ -423,8 +423,17 @@ Adds time series data from a metadata file or metadata descriptors.
   that includes an array of IS.TimeSeriesFileMetadata instances or a vector.
 - `resolution::DateTime.Period=nothing`: skip time series that don't match this resolution.
 """
-function add_time_series!(sys::System, metadata_file::AbstractString; resolution = nothing)
-    return IS.add_time_series!(Component, sys.data, metadata_file; resolution = resolution)
+function add_time_series_from_file_metadata!(
```
Comment (jd-lara) [inline src/base.jl:426]: Can we keep the previous name? I think we can leave the details underneath

File: src/base.jl [hunk 2]
```diff
@@ -938,34 +895,29 @@ Return a TimeSeriesData for a subset of the time series range stored for these p
 function get_time_series(
     ::Type{T},
     component::Component,
-    initial_time::Dates.DateTime,
-    label::AbstractString,
+    name::AbstractString,
```
Comment (jd-lara) [inline src/base.jl:898]: The get_time_series method in IS allows passing an initial time and horizon here.

File: src/base.jl [hunk 3]
```diff
@@ -974,11 +926,19 @@ Return a TimeSeries.TimeArray for the requested time series parameters.
...
-    initial_time::Dates.DateTime,
-    label::AbstractString,
-    horizon::Union{Nothing, Int} = nothing,
+    name::AbstractString;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+    count::Int = 1,
 ) where {T <: TimeSeriesData}
-    return IS.get_time_series_array(T, component, initial_time, label, horizon)
+    return IS.get_time_series_array(
```
Comment (jd-lara) [inline src/base.jl:934]: Do we need to implement this for Deterministic?

File: src/base.jl [hunk 4]
```diff
@@ -1008,11 +976,19 @@ Return an Array of values for the requested time series parameters.
...
-    initial_time::Dates.DateTime,
-    label::AbstractString,
-    horizon::Union{Nothing, Int} = nothing,
+    name::AbstractString;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+    count::Int = 1,
 ) where {T <: TimeSeriesData}
-    return IS.get_time_series_values(T, component, initial_time, label, horizon)
+    return IS.get_time_series_values(
```
Comment (jd-lara) [inline src/base.jl:984]: Same here, do we have this for Deterministic?

**Review by jd-lara — APPROVED** [review]
The changes look good. Which methods are we missing here for Deterministic?

File: src/base.jl (hunk 2 above)
Comment (daniel-thom) [reply src/base.jl:898]: Agree. Must fix.

File: src/base.jl (hunk 3 above)
Comment (daniel-thom) [reply src/base.jl:934]: I think this supports Deterministic as it is.  Why do you disagree?

File: src/base.jl (hunk 4 above)
Comment (daniel-thom) [reply src/base.jl:984]: Same as above

File: src/base.jl (hunk 3 above)
Comment (jd-lara) [reply src/base.jl:934]: I am just inquiring to see if we need to test it.

Comment (daniel-thom) [reply src/base.jl:934]: We should.  Can you add some test data to PowerSystems or PowerSystemsTestData?

Comment (jd-lara) [reply src/base.jl:934]: Ok, I will make sure to make/add data for this test. I am almost sure that the 5-bus case already has data on this effect.

**Pull request merged**